ocaml/idl/datamodel_vm_group.ml

```diff
@@ -0,0 +1,67 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let placement_type =
+  Enum
+    ( "vm_placement_type"
+    , [
+        ("anti_affinity", "VM anti-affinity group")
+      ; ("normal", "Normal type group")
+      ]
+    )
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Create a new VM group"
+    ~params:
+      [
+        (String, "name_label", "The name of the VM group")
+      ; (String, "name_description", "The description of the VM group")
+      ; (placement_type, "placement", "The placement type of the VM group")
+      ]
+    ~result:(Ref _vm_group, "The ref of the created VM group record")
+    ~allowed_roles:_R_VM_ADMIN ()
+
+let destroy =
+  call ~name:"destroy" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Remove the VM group record from the database"
+    ~params:
+      [
+        ( Ref _vm_group
+        , "self"
+        , "The Ref of the VM group to be removed from the database"
+        )
+      ]
+    ~allowed_roles:_R_VM_ADMIN ()
```

Comment:
Can we use `~gen_constructor_destructor:true` below and delete these two explicit function definitions?